doc-content/jbpm-docs/src/main/asciidoc/BPMN2/con_custom-tasks-overview.adoc

```diff
@@ -0,0 +1,216 @@
+
+You can click *Assignments* in the *Properties* panel to open the *Milestone Data I/O* window. In the *Milestone Data I/O* window, you can configure the data input and output as required. For example, to execute a milestone, enter the following data inputs in *Data Inputs and Assignments* fields:
+
+* *Condition*: Condition for the milestone to meet. For example, you can enter a Java expression (string data type) that uses a process variable.
+
+.DecisionTask
+A decision task is used to execute a DMN diagram and invoke a decision engine service from a process. By default, a decision task maps to the DMN decision.
+
+image::BPMN2/bpmn-decision-task-custom.png[]
+
+You can use decision tasks to make an operational decision in a process. Decision tasks are useful for identifying key decisions in a process that need to be made.
+
+You can click *Assignments* in the *Properties* panel to open the *Decision Task Data I/O* window. In the *Decision Task Data I/O* window, you can configure the data input and output as required. For example, to execute a decision task, enter the following data inputs in *Data Inputs and Assignments* fields:
+
+* *Decision*: Decision for a process to make.
+* *Language*: Language of the DMN model, such as `FEEL` or `S-FEEL`.
+* *Model*: Name of the DMN model.
+* *Namespace*: Namespace of the DMN model.
+
+.BusinessRuleTask
+A business rule task is used to evaluate a DRL rule and invoke a decision engine service from a process. By default, a business rule task maps to the DRL rules.
+
+image::BPMN2/bpmn-business-rule-custom-task.png[]
+
+You can use business rule tasks to evaluate key business rules in a business process. You can click *Assignments* in the *Properties* panel to open the *Business Rule Task Data I/O* window. In the *Business Rule Task Data I/O* window, you can configure the data input and output as required. For example, to execute a business rule task, enter the following data inputs in *Data Inputs and Assignments* fields:
+
+* *KieSessionName*: Name of the KIE session.
+* *KieSessionType*: Type of the KIE session.
+* *Language*: Language of the business rule.
+
+.KafkaPublishMessages
+A Kafka work item is used to send events to a Kafka topic. This custom task includes a work item handler, which uses the Kafka producer to send messages to a specific Kafka server topic. For example, `KafkaPublishMessages` task publishes messages from a process to a Kafka topic.
+
+image::BPMN2/bpmn-kafkapublishmessages-task.png[]
+
+You can click *Assignments* in the *Properties* panel to open the *KafkaPublishMessages Data I/O* window. In the *KafkaPublishMessages Data I/O* window, you can configure the data input and output as required. For example, to execute a Kafka work item, enter the following data inputs in *Data Inputs and Assignments* fields:
+
+* *Key*: Unique Id of Kafka topic.
```

Comment:
This is a key of a Kafka message to be sent.

Reply:
Done, changed to _Key of the Kafka message to be sent_